src/shared/lib/validateParameters.ts

```diff
@@ -0,0 +1,25 @@
+import { validate, ParameterValidationError } from 'parameter-validator';
+import * as _ from 'lodash';
+
+export interface URLValidationResult {
+    isValid:Boolean;
+    message:string | null
+}
+
+export default function(params: { [k:string]: string }, rules:any): URLValidationResult {
```

Comment:
good catch.  I have the rules hardcoded in call to the validate-parameters library.  The rules are the validation rules configuration accepted by library.  The reason I decided to wrap the library was that I wanted zero length string to be considered invalid.